scipy/spatial/distance.py

```diff
@@ -746,7 +746,9 @@ def hamming(u, v, w=None):
         w = _validate_weights(w)
         if w.shape != u.shape:
             raise ValueError("'w' should have the same length as 'u' and 'v'.")
-    return np.average(u_ne_v, weights=w)
+        w /= w.sum()
```

Comment:
> is it slower to use np.sum(w)?

Looks like it's a bit slower probably due to some constant overhead. I just opted for `.sum()` because I think that's what `np.average` uses internally as well.
```py
In [10]: w = np.random.uniform(size=1024)

In [11]: %timeit w.sum()
1.49 µs ± 16.1 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)

In [12]: %timeit np.sum(w)
2.62 µs ± 38.7 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
```

Reply:
Thanks for checking, that's fine then. In the long run, we're likely to swap out for `xp.sum` to make use of [the array API](https://data-apis.org/array-api/latest/API_specification/index.html), but may as well stick with the faster version for now 👍